diag_manager/diag_output.F90

```diff
@@ -1388,4 +1389,957 @@ SUBROUTINE set_diag_global_att(component, gridType, tileName)
   END SUBROUTINE set_diag_global_att
   ! </SUBROUTINE>
 
+!END MODULE diag_output_mod
+#else
```

Comment:
Please combine the two parts as you have done in `diag_manager.F90`

Reply:
The routines in this file are so vastly different, that I think it makes the most sense to keep them separate like this.  In the next FMS release, we are working on renaming the routines that use mpp_io so that they will called differently.  If something needs to be patched, it's likely to only need to be patched in one place in this module.

Reply:
If that is the plan, then I suggest placing the `#if`/`#endif` around the routines, modifying each just enough so both can use the same file header stuff (e.g. `use`, `#include`, etc.).

I also noticed in the code places where instead of deleting lines that are not needed, they are commented out (but no comment as to why they are kept).  I also have seen things like the following:
```fortran
IF ( <some_test> ) THEN
  ! Do nothing
ELSE
  STATEMENT1
  STATEMENT2
END IF
```

Using an `IF` like this causing issues later, as it is difficult to understand _why_ the `IF` tests for something that has no work.  These cases should be negated:

```fortran
IF ( .NOT.<some_test> ) THEN
  STATEMENT1
  STATEMENT2
END IF.
```

Easier to understand and maintain.

Reply:
The two different code bases use different `#include`s, so I think this is actually messier now.  I did it though. 
 
I think the clean up (ifs, commented code) could/should be done on a different PR since it is a separate issue.